.circleci/config.yml

```diff
@@ -13,26 +112,215 @@ jobs:
             python -m venv ~/venv
             echo 'export PATH=~/venv/bin:$PATH' >> ${BASH_ENV}
       - restore_cache:
-          keys:
-          - python-deps-{{ checksum "setup.py" }}-{{ checksum "requirements-dev.txt" }}
-          - python-deps-
-      - run:
-          name: Install lint requirements
-          command: pip install -r requirements-dev.txt
-      - run:
-          name: Run linters
-          command: tox -e lint
-      - run:
-          name: Install project
-          command: pip install -e .
-      - run:
-          name: Run Unit-tests
-          command: tox
+          <<: *cache_load_key
+
+  # Save our installed dependencies to the cache.
+  save-deps-to-cache:
+    <<: *job_defaults
+    steps:
       - save_cache:
-          key: python-deps-{{ checksum "setup.py" }}-{{ checksum "requirements-dev.txt" }}
+          <<: *cache_save_key
```

Comment:
I like passing variables, and I like defining them at the top of a file. And while the whole "pipelines as code" mantra would make me consider this code, I can agree with you here. Especially since they(this and below) are only used once.